core/opendaq/device/include/opendaq/log_file_info.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022-2024 openDAQ d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <coretypes/common.h>
+#include <coretypes/stringobject.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+struct ILogFileInfoBuilder;
+
+/*!
+ * @ingroup opendaq_log_file_info
+ * @addtogroup opendaq_log_file_info log_file_info
+ * @{
+ */
+
+enum class LogFileEncodingType: uint32_t
```

Comment:
I'm not sure if enum is a good solution for the encoding. Why not just string. It allows any encoding? Discuss this with Jaka!
[https://www.iana.org/assignments/character-sets/character-sets.xhtml](url)